minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java

```diff
@@ -522,6 +531,42 @@ public synchronized void stopMaster(int i) throws IOException {
     mMasters.get(i).close();
   }
 
+  /**
+   * Removes master i from the cluster.
+   * @param i index of the master to be removed
+   * @throws IOException
+   */
+  public synchronized void removeMaster(int i) throws IOException {
+    stopMaster(i);
+    mMasterAddresses.remove(i);
+    mMasterIds.remove(i);
+    mMasters.remove(i);
+    mNumMasters--;
+    if (mMasterProperties.containsKey(i)) {
+      mMasterProperties.remove(i);
+    }
```

Comment:
technically an unecessary check because `Map#remove` will just return null if the key doesn't exist.

Reply:
fixed